pydantic/main.py

```diff
@@ -687,8 +687,10 @@ def validate_model(  # noqa: C901 (ignore complexity)
                     errors.append(ErrorWrapper(ExtraError(), loc=f, config=config))
 
     if not raise_exc:
-        return values, fields_set, ValidationError(errors) if errors else None
+        model_type = model if isinstance(model, type) else type(model)
+        return values, fields_set, ValidationError(errors, model_type) if errors else None
 
     if errors:
-        raise ValidationError(errors)
+        model_type = model if isinstance(model, type) else type(model)
```

Comment:
@samuelcolvin Any preference for how to eliminate the duplicated definition of model_type? I could move it outside the conditionals but I didn't want to affect performance in the case where an error is not raised.

Reply:
we're also executing `model_type = ...` above even when it's not required.

Maybe something like

```py
err = None
if errors:
    model_type = model if isinstance(model, type) else type(model)
    err = ValidationError(errors, model_type)

if not raise_exc:
    return values, fields_set, err

if err:
    raise err
```
?